tests/Unit/GeneralizedReferrerTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace WMDE\Fundraising\Tests\Unit;
+
+use WMDE\Fundraising\Frontend\GeneralizedReferrer;
+use WMDE\Fundraising\Frontend\Tests\Unit\Validation\ValidatorTestCase;
+
+/**
+ * @covers WMDE\Fundraising\Frontend\GeneralizedReferrer
+ *
+ * @licence GNU GPL v2+
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
+ */
+class GeneralizedReferrerTest extends ValidatorTestCase {
+
+	private $domainMap = [
+		'wikimedia.de' => 'wikimedia.de',
+		'www.wikimedia.de' => 'wikimedia.de',
+		'wikipedia.de' => 'wikipedia.de',
+		'www.wikipedia.de' => 'wikipedia.de',
+		'de.wikipedia.org' => 'de.wikipedia.org',
+		'en.wikipedia.org' => 'en.wikipedia.org',
+		'ru.wikivoyage.org' => 'ru.wikivoyage.org',
+	];
+
+	public function urlProvider() {
+		return [
```

Comment:
Personally I think it makes more sense to put the provider after the test method that uses it